packages/react-positioning/src/usePopper.ts

```diff
@@ -401,30 +231,48 @@ export function usePopper(
     if (options.target) {
       overrideTargetRef.current = options.target;
     }
-  }, [options.target, overrideTargetRef]);
-  useIsomorphicLayoutEffect(() => {
-    handlePopperUpdate();
+  }, [options.target, overrideTargetRef, containerRef]);
 
-    return () => {
-      popperInstanceRef.current?.destroy();
-      popperInstanceRef.current = null;
-    };
-  }, [handlePopperUpdate, options.enabled]);
+  useIsomorphicLayoutEffect(
+    () => {
+      updatePosition();
+    },
+    // Missing deps:
+    // updatePosition - referentially stable
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [enabled],
+  );
```

Comment:
IMO, it's easier to add `updatePosition` to deps than describe why it's disabled